process.go

```diff
@@ -0,0 +1,179 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"io"
+
+	"github.com/urfave/cli"
+)
+
+// == Direct input processing ========================================
+
+func processInputFiles(c *cli.Context, g *Gomplate) error {
+	inputs, err := readInputs(c.String("in"), c.StringSlice("file"))
+	if err != nil {
+		return err
+	}
+
+	outputs := c.StringSlice("out")
+	if len(outputs) == 0 {
+		outputs = []string{"-"}
+	}
+
+	for n, input := range inputs {
+		if err := renderTemplate(g, input, outputs[n]); err != nil {
```

Comment:
It'd be ideal if this were done in parallel, using goroutines - I think it should be pretty straightforward...